toolchain/lex/lex.cpp

```diff
@@ -967,29 +1014,24 @@ auto Lexer::LexNumericLiteral(llvm::StringRef source_text, ssize_t& position)
   return VariantMatch(
       literal->ComputeValue(emitter_),
       [&](NumericLiteral::IntValue&& value) {
-        auto token = buffer_.AddToken(
-            {.kind = TokenKind::IntLiteral, .byte_offset = byte_offset});
-        buffer_.GetTokenInfo(token).int_id =
-            buffer_.value_stores_->ints().Add(std::move(value.value));
-        return token;
+        return LexToken(
+            TokenKind::IntLiteral,
+            buffer_.value_stores_->ints().Add(std::move(value.value)).index,
+            byte_offset);
       },
       [&](NumericLiteral::RealValue&& value) {
-        auto token = buffer_.AddToken(
-            {.kind = TokenKind::RealLiteral, .byte_offset = byte_offset});
-        buffer_.GetTokenInfo(token).real_id =
-            buffer_.value_stores_->reals().Add(Real{
-                .mantissa = value.mantissa,
-                .exponent = value.exponent,
-                .is_decimal = (value.radix == NumericLiteral::Radix::Decimal)});
-        return token;
+        return LexToken(
+            TokenKind::RealLiteral,
+            buffer_.value_stores_->reals()
+                .Add(Real{.mantissa = value.mantissa,
+                          .exponent = value.exponent,
+                          .is_decimal =
+                              (value.radix == NumericLiteral::Radix::Decimal)})
+                .index,
+            byte_offset);
```

Comment:
```suggestion
        auto real_id = buffer_.value_stores_->reals().Add(Real{
            .mantissa = value.mantissa,
            .exponent = value.exponent,
            .is_decimal = (value.radix == NumericLiteral::Radix::Decimal)});
        return LexToken(TokenKind::RealLiteral, real_id.index, byte_offset);
```

Small suggestion, I think separating the add may make this easier to read.